shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/BridgeIngressServiceImpl.java

```diff
@@ -77,6 +77,12 @@ public void createBridgeIngress(BridgeDTO bridgeDTO) {
 
             // create or update the secrets for the bridgeIngress
             createOrUpdateBridgeIngressSecret(bridgeIngress, bridgeDTO);
+        } else {
+            LOGGER.info("Existing BridgeIngress '{}' matched the expected specification. No update required. Notifying manager that is it ready.", bridgeDTO.getId());
+            bridgeDTO.setStatus(ManagedResourceStatus.READY);
+            managerClient.notifyBridgeStatusChange(bridgeDTO).subscribe().with(
+                    success -> LOGGER.debug("Ready notification for BridgeIngress '{}' has been sent to the manager successfully", bridgeDTO.getId()),
+                    failure -> LOGGER.error("Failed to send updated status to Manager for entity of type '{}'", BridgeDTO.class.getSimpleName(), failure));
```

Comment:
well this is not always true. 
Consider the following scenario: 
1) Something goes wrong and the resource is in a FAILED state. 
2) This method is called and there is no change in the CRD
3) -> with the current implementation you update the manager with the READY state, while the resource is in a FAILED state. 

In short, I think you should return the status you find in the `existing` resource.

Reply:
Hi @r00ta the best I can do in the `operator` (without significant refactoring) is `processorDTO.setStatus(existing.getStatus().isReady() ? ManagedResourceStatus.READY : ManagedResourceStatus.FAILED)`... i.e. if the CRD was "ready" I signal the `manager` that the `ManagedResource` is `READY` otherwise it must have `FAILED`. The CRD controller only sets the `ManagedResourceStatus` to either `READY` or `FAILED` so it seems reasonable. WDYT?

Reply:
@r00ta given @robpblake [comment](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/641#issuecomment-1102861125) IDK whether I should remove the check from `manager` as given your insightful comment if a CRD was `FAILED` and I do the check in the `manager` (as Rob proposed and I've implemented) we'll never get the operator to try and re-provision the updated definition... unless I also check if the state is `READY`.. hmm.. let me add that as a commit until we've finalised our discussion.